app/Console/AbstractGeocodeCommand.php

```diff
@@ -15,10 +16,24 @@ abstract class AbstractGeocodeCommand extends AbstractCommand
      */
     private $searchResultsGenerator;
 
-    public function __construct(CommandBusInterface $commandBus, ResultsGeneratorInterface $searchResultsGenerator)
-    {
+    /**
+     * @var DocumentRepository
+     */
+    private $documentRepository;
+
+    public function __construct(
+        CommandBusInterface $commandBus,
+        ResultsGeneratorInterface $searchResultsGenerator,
+        DocumentRepository $documentRepository
+    ) {
         parent::__construct($commandBus);
         $this->searchResultsGenerator = $searchResultsGenerator;
+        $this->documentRepository = $documentRepository;
+    }
+
+    protected function getDocumentRepository(): DocumentRepository
```

Comment:
Why not expose the property itself as protected?

Reply:
I think either way is fine personally, no strong preference. I guess it's just a habit of making as much as possible private and only exposing what's needed (in this case returning the repository, but not being able to overwrite it). Anyway, it doesn't really matter since we refactored it a commit or two later to not expose the repository at all but to add a method to directly get the needed documents.